### PR TITLE
feat: home page with features like new Videos

### PR DESCRIPTION
**Pull request by vrx29**
### Gimesto Play : video library like youtube

Live Preview : https://gimesto-play-git-home-vrx29.vercel.app/

Some files are images and icons and few are CSS files I you want to check the main files only then check these files.
PRs to be checked:

:hear_no_evil: [Banner](https://github.com/vrx29/gimesto-play/pull/1/files#diff-518a900cf83a868d6bb887b88907ad6c57dbd0229fe2140e2587651be3eca800)
:hear_no_evil: [header](https://github.com/vrx29/gimesto-play/pull/1/files#diff-6efde9512356a2f5c3cc4c161365127d269413d13ae8fb43c3395c334981f447)
:hear_no_evil: [Video Card](https://github.com/vrx29/gimesto-play/pull/1/files#diff-267784b43cf07d3345b7c9c82bd286a36b24093d602663d29ee475d1d689caaa)
:hear_no_evil: [Video Context](https://github.com/vrx29/gimesto-play/pull/1/files#diff-a5022f1409674001e58ff7ba0919d901a3ddc026f0c9a9fa0b4be5fab2233587)
:hear_no_evil: [Fetch hook](https://github.com/vrx29/gimesto-play/pull/1/files#diff-697611df152fff126d2a1d2be956d7e15ba3651aa951be30351b548351f596af)
:hear_no_evil: [sharedReducer](https://github.com/vrx29/gimesto-play/pull/1/files#diff-537619cdf82273c75c0616885f7ee58d3f0f3ca6fd26ddb8c70b17de3e2fd5aa)

### CODE REVIEW

**Review by animeshgarai09 — APPROVED** [review]
The codebase is well structured with amazing UI 🔥

Icons can be imported as a react component, no need to create a separate file.

File: src/reducers/reducerFunctions/sharedReducer.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import {
+  RESET_LOADING,
+  SET_ERROR,
+  SET_LOADING,
+  SET_SUCCESS,
+} from "reducers/constants";
+
+export const sharedReducer = (state, { type, payload }) => {
+  switch (type) {
+    case SET_LOADING:
```
Comment (yash9263) [inline src/reducers/reducerFunctions/sharedReducer.js:10]: when the function is called you can set error to null to clear out the previous error states

File: src/reducers/reducerFunctions/sharedReducer.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      return { ...state, loading: true };
+    case RESET_LOADING:
+      return { ...state, loading: false };
+    case SET_SUCCESS:
+      return { ...state, data: payload };
```
Comment (yash9263) [inline src/reducers/reducerFunctions/sharedReducer.js]: after the data is loaded set the error to null.

**Comment by yash9263** [comment]
The code is clean and clear easy to understand.
Nice use of custom hooks and useReducer to handle state for async function.
One improved you can add is like instead of using SET_LOADING, RESET_LOADING  you can have your 
switch case like
 ```js
case 'loading': 
  return { loading: true, error: null, data: null }
case 'error':
  return { loading: false, error: value, data: null }
case 'success':
  return { loading: false, error: null, data: value }
```

**Review by yash9263 — APPROVED** [review]

**Pull request merged**